cli/kubernetes/watch.go

```diff
@@ -28,56 +28,34 @@ func FilteredWatch(ctx context.Context, kubernetesProvider *Provider, targetName
 
 		go func(targetNamespace string) {
 			defer wg.Done()
-			watchRestartDebouncer := debounce.NewDebouncer(5 * time.Second, func() {})
-			shouldStop := false
+			watchRestartDebouncer := debounce.NewDebouncer(1 * time.Minute, func() {})
 
-			for !shouldStop {
+			for {
 				watcher := kubernetesProvider.GetPodWatcher(ctx, targetNamespace)
-				func() {
-					for !shouldStop {
-						select {
-						case e := <-watcher.ResultChan():
-							if e.Type == watch.Error {
-								errorChan <- apierrors.FromObject(e.Object)
-								shouldStop = true
-								return
-							}
-							if e.Object == nil {
-								if !watchRestartDebouncer.IsOn() {
-									logger.Log.Debug("detected a potential harmless watch timeout, retrying watch loop")
-									return
-								} else {
-									errorChan <- errors.New("received too many unknown errors in k8s watch")
-									shouldStop = true
-									return
-								}
-
-							}
-
-							pod, ok := e.Object.(*corev1.Pod)
-							if !ok {
-								continue
-							}
-
-							if !podFilter.MatchString(pod.Name) {
-								continue
-							}
-
-							switch e.Type {
-							case watch.Added:
-								addedChan <- pod
-							case watch.Modified:
-								modifiedChan <- pod
-							case watch.Deleted:
-								removedChan <- pod
-							}
-						case <-ctx.Done():
-							watcher.Stop()
-							shouldStop = true
-							return
-						}
+				err := startWatchLoop(ctx, watcher, podFilter, addedChan, modifiedChan, removedChan) // blocking
+				watcher.Stop()
+
+				select {
+				case <- ctx.Done():
+					return
+				default:
+					break
+				}
+
+				if err != nil {
+					errorChan <- errors.New("received too many unknown errors in k8s watch")
```

Comment:
Include information about the original error, otherwise it would be hard to debug from logs.
```go
errorChan <- fmt.Errorf("error in k8 watch: %w", err)
```

Reply:
whoops